tests/ecmp/test_fgnhg.py

```diff
@@ -318,25 +314,99 @@ def common_setup_teardown(tbinfo, duthosts, rand_one_dut_hostname):
 
     try:
         mg_facts   = duthost.get_extended_minigraph_facts(tbinfo)
+        #host_facts = duthost.setup()['ansible_facts']
         cfg_facts = duthost.config_facts(host=duthost.hostname, source="persistent")['ansible_facts']
         router_mac = duthost.facts['router_mac']
         net_ports = []
         for name, val in mg_facts['minigraph_portchannels'].items():
             members = [mg_facts['minigraph_ptf_indices'][member] for member in val['members']]
             net_ports.extend(members)
-        yield duthost, cfg_facts, router_mac, net_ports 
+        yield duthost, cfg_facts, router_mac, net_ports
 
     finally:
         cleanup(duthost)
 
 
-def test_fg_ecmp(common_setup_teardown, ptfadapter, ptfhost):
+def test_fg_ecmp(tbinfo, common_setup_teardown, ptfadapter, ptfhost):
     duthost, cfg_facts, router_mac, net_ports = common_setup_teardown
 
-    # IPv4 test
-    port_list, ip_to_port, bank_0_port, bank_1_port = setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts, router_mac, net_ports, DEFAULT_VLAN_IPv4, PREFIX_IPv4)
-    fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, PREFIX_IPv4) 
+    if MINIGRAPH_PARSING == "OFF":
+        # IPv4 test
+        port_list, ip_to_port, bank_0_port, bank_1_port = setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts,
+                                                                            router_mac, net_ports, DEFAULT_VLAN_IPv4,
+                                                                            PREFIX_IPv4)
+        fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, PREFIX_IPv4)
 
-    # IPv6 test
-    port_list, ip_to_port, bank_0_port, bank_1_port = setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts, router_mac, net_ports, DEFAULT_VLAN_IPv6, PREFIX_IPv6)
-    fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, PREFIX_IPv6) 
+        # IPv6 test
+        port_list, ip_to_port, bank_0_port, bank_1_port = setup_test_config(ptfadapter, duthost, ptfhost, cfg_facts,
+                                                                            router_mac, net_ports, DEFAULT_VLAN_IPv6,
+                                                                            PREFIX_IPv6)
+        fg_ecmp(ptfhost, duthost, router_mac, net_ports, port_list, ip_to_port, bank_0_port, bank_1_port, PREFIX_IPv6)
+
+    else:
+        logger.info("Minigraph Parsing Test Case Running...")
+        mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+        prefix_ipv4 = ""
+        prefix_ipv6 = ""
+        bank_0_port_ipv4 = []
+        bank_1_port_ipv4 = []
+        bank_0_port_ipv6 = []
+        bank_1_port_ipv6 = []
+        ipv4_to_port = {}
+        ipv6_to_port = {}
```

Comment:
Can we move the parsing and config obtaining part to a different function, ie: line 347 to 394 in a separate function and return the relevant values from that function?

Reply:
Sure will do that